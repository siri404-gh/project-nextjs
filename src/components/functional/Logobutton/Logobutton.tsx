import React, { FunctionComponent } from 'react';
import { makeStyles } from '@material-ui/core';
import cn from 'classnames';

import monochrome from '@/assets/monochrome_icon.png';
import dichrome from '@/assets/icon-56.png';

const useStyles = makeStyles(() => ({
  logo: {
    cursor: 'pointer',
    height: 32,
    width: 32,
    padding: 2,
  },
}));

export interface Props {
  className?: string,
  invert?: boolean,
}

const LogoButton: FunctionComponent<Props> = ({ className, invert }) => {
  const classes = useStyles();
  return (
    <img alt='logo' className={cn(className, classes.logo)} src={invert ? monochrome : dichrome} />
  );
};

LogoButton.defaultProps = {
  className: '',
  invert: false,
};

export default LogoButton;
