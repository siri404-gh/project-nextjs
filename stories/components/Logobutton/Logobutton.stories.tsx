import React from 'react';
import { Story } from '@storybook/react/types-6-0';

import Logobutton, { Props } from '@/components/functional/Logobutton/Logobutton';

export default {
  title: 'Logobutton',
  component: Logobutton,
};

const Template: Story<Props> = args => <Logobutton {...args} />;

export const FirstStory = Template.bind({});

FirstStory.args = {};
