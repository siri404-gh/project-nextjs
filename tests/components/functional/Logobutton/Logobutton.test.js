import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Logobutton from '@/components/functional/Logobutton/Logobutton';

describe('Logobutton', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('section');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render without props', () => {
    act(() => {
      render(<Logobutton />, container);
    });
    expect(container).toMatchSnapshot();
  });

  it('Should render inverted logo', () => {
    act(() => {
      render(<Logobutton invert />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
