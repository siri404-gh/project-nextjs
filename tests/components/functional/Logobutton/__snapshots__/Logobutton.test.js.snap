// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logobutton Should render inverted logo 1`] = `
<section>
  <img
    alt="logo"
    class="makeStyles-logo-2"
    src="test-file-stub"
  />
</section>
`;

exports[`Logobutton Should render without props 1`] = `
<section>
  <img
    alt="logo"
    class="makeStyles-logo-1"
    src="test-file-stub"
  />
</section>
`;
